### PR TITLE
Wait for fresh nns after reboot at test

### PR DESCRIPTION
**Pull request by qinqon**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
/kind test

**What this PR does / why we need it**:
the default brindge and bond e2e tests were reading not
fresh nns after reboot, they have to check that update timestamp
is fresh enough to read information from there if not they will
face a false possitive.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: automation/check-patch.e2e-ocp.sh [hunk 1]
```diff
@@ -1,33 +0,0 @@
-#!/bin/bash -xe
```
Comment (phoracek) [inline automation/check-patch.e2e-ocp.sh:1]: We indeed should drop this burden (and from infra as well). However, do it please in a different PR

File: automation/check-patch.e2e-k8s.sh [hunk 2]
```diff
@@ -11,7 +11,7 @@ teardown() {
     make cluster-down
     cp $(find . -name "*junit*.xml") $ARTIFACTS
     # Don't fail if there is no logs
-    cp ${E2E_LOGS}/*.log ${ARTIFACTS} || true
+    cp ${E2E_LOGS}/handler/*.log ${ARTIFACTS} || true
```
Comment (phoracek) [inline automation/check-patch.e2e-k8s.sh:14]: Not related to the PR

File: test/e2e/handler/default_bridged_network_test.go [hunk 3]
```diff
@@ -141,7 +141,7 @@ var _ = Describe("NodeNetworkConfigurationPolicy default bridged network", func(
 
 				err := restartNode(nodeToReboot)
 				Expect(err).ToNot(HaveOccurred())
-
+				waitForFreshNodeNetworkState(nodeToReboot)
```
Comment (phoracek) [inline test/e2e/handler/default_bridged_network_test.go:144]: Please add empty lines around it

File: test/e2e/handler/bonding_default_interface_test.go [hunk 4]
```diff
@@ -104,6 +104,10 @@ var _ = Describe("NodeNetworkConfigurationPolicy bonding default interface", fun
 			By(fmt.Sprintf("Reboot node %s and verify that bond still has ip of primary nic", nodeToReboot))
 			err := restartNode(nodeToReboot)
 			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("wait for nns to be refreshed at %s", nodeToReboot))
```
Comment (phoracek) [inline test/e2e/handler/bonding_default_interface_test.go:108]: We start `By`s with capital letter

File: test/e2e/handler/utils.go [hunk 5]
```diff
@@ -298,6 +298,17 @@ func interfacesForNode(node string) AsyncAssertion {
 	}, ReadTimeout, ReadInterval)
 }
 
+func waitForFreshNodeNetworkState(node string) {
+	// Worst case is 5 seconds so let's just use time.now()
```
Comment (phoracek) [inline test/e2e/handler/utils.go:302]: You are commenting on an issue that is not visible in the code (at least not at this line). Could you please elaborate what is the issue in more detail? I don't see an issue using `Now`

File: test/e2e/handler/utils.go [hunk 6]
```diff
@@ -298,6 +298,17 @@ func interfacesForNode(node string) AsyncAssertion {
...
+	now := time.Now()
+	EventuallyWithOffset(1, func() time.Time {
+		key := types.NamespacedName{Namespace: framework.Global.Namespace, Name: node}
+		nnsUpdateTime := nodeNetworkState(key).Status.LastSuccessfulUpdateTime
+		return nnsUpdateTime.Time
+	}, 4*time.Minute, 5*time.Second).Should(BeTemporally(">=", now), fmt.Sprintf("Node %s should has a fresh nns)", node))
```
Comment (phoracek) [inline test/e2e/handler/utils.go:308]: It should be "should have"

File: test/e2e/handler/utils.go [hunk 7]
```diff
@@ -298,6 +298,17 @@ func interfacesForNode(node string) AsyncAssertion {
 	}, ReadTimeout, ReadInterval)
 }
 
+func waitForFreshNodeNetworkState(node string) {
```
Comment (phoracek) [inline test/e2e/handler/utils.go:301]: I think it is more correct to say `waitForNodeNetworkStateUpdate`

File: automation/check-patch.e2e-k8s.sh (hunk 2 above)
Comment (qinqon) [reply automation/check-patch.e2e-k8s.sh:14]: Well without prow artifacts it's difficult to do forensics at prow.
